cmd/infra/powervs/create.go

```diff
@@ -1292,10 +1321,13 @@ func (infra *Infra) isCloudConnectionReady(ctx context.Context, options *CreateI
 		return err
 	}
 
+	var retryCount int
 	f := func() (bool, error) {
 		cloudConn, err = client.Get(infra.CloudConnectionID)
 		if err != nil {
-			return false, err
+			if err = isRetryableError(err, timeoutErrorKeywords, &retryCount); err != nil {
+				return false, err
+			}
```

Comment:
If its hitting 522 or 524 then it should return function `false, nil` and go for next iteration right instead of proceeding further?

Reply:
yeah that's the intention, I have refactored to look more clear, ptal.